data-prepper-plugins/geoip-prepper/src/main/java/com/amazon/dataprepper/plugins/prepper/geoip/GeoIpPrepper.java

```diff
@@ -72,7 +72,9 @@ public GeoIpPrepper(final PluginSetting pluginSetting) {
                     if (foundData.isPresent()) {
                         Map<String, Object> locationDataMap = OBJECT_MAPPER.convertValue(foundData.get(), MAP_TYPE_REFERENCE);
                         for (Map.Entry<String, Object> entry : locationDataMap.entrySet())
-                            rawSpanMap.put(entry.getKey(), entry.getValue());
+                            if (entry.getValue() != null) {
```

Comment:
Are we seeing null values in the location data map ?

Reply:
This is for when users only choose a few fields that they want from the GeoIP lookup. E.g. if they only want city/state, previously, every other field would show up as null. Now, we only attach the fields that were requested and returned.